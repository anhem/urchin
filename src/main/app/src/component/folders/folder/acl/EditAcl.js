import React, {Component} from 'react'

class EditAcl extends Component {

    updateAclForUser = (userId, e) => {
        this.props.callbacks.updateAclForUser(userId, e.target.name, e.target.checked);
    };

    updateAclForGroup = (groupId, e) => {
        this.props.callbacks.updateAclForGroup(groupId, e.target.name, e.target.checked);
    };

    render() {
        let usersAcl = this.props.aclUsers || [];
        let groupsAcl = this.props.aclGroups || [];

        let userItems = usersAcl.map((item, index) =>
            <div className="edit-acl__users__user"
                 key={index.toString()}>
                <div data-view={item.username}>{item.username}</div>
                <div>
                    <label><input data-view={`${item.username}-read`}
                                  type="checkbox"
                                  name="read"
                                  checked={item.read}
                                  onChange={(e) => this.updateAclForUser(item.userId, e)}
                    />
                        read
                    </label>
                </div>
                <div>
                    <label>
                        <input data-view={`${item.username}-write`}
                               type="checkbox"
                               name="write"
                               checked={item.write}
                               onChange={(e) => this.updateAclForUser(item.userId, e)}
                        />
                        write
                    </label>
                </div>
                <div>
                    <label>
                        <input data-view={`${item.username}-execute`}
                               type="checkbox"
                               name="execute"
                               checked={item.execute}
                               onChange={(e) => this.updateAclForUser(item.userId, e)}
                        />
                        execute
                    </label>
                </div>
            </div>
        );

        let groupItems = groupsAcl.map((item, index) =>
            <div className="edit-acl__groups__group"
                 key={index.toString()}>
                <div data-view={item.groupName}>{item.groupName}</div>
                <div>
                    <label>
                        <input data-view={`${item.groupName}-read`}
                               type="checkbox"
                               name="read"
                               checked={item.read}
                               onChange={(e) => this.updateAclForGroup(item.groupId, e)}
                        />
                        read
                    </label>
                </div>
                <div>
                    <label>
                        <input data-view={`${item.groupName}-write`}
                               type="checkbox"
                               name="write"
                               checked={item.write}
                               onChange={(e) => this.updateAclForGroup(item.groupId, e)}
                        />
                        write
                    </label>
                </div>
                <div>
                    <label>
                        <input data-view={`${item.groupName}-execute`}
                               type="checkbox"
                               name="execute"
                               checked={item.execute}
                               onChange={(e) => this.updateAclForGroup(item.groupId, e)}
                        />
                        execute
                    </label>
                </div>
            </div>
        );

        return (
            <div data-view="editAcl"
                 className="edit-acl">
                <h2>Group permissions</h2>
                <div className="edit-acl__groups">
                    {groupItems}
                </div>
                <h2>User permissions</h2>
                <div className="edit-acl__users">
                    {userItems}
                </div>
            </div>

        )
    }
}

export default EditAcl