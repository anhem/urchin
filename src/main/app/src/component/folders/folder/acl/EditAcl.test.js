import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import EditAcl from "./EditAcl";

describe('EditAcl', () => {
    const userId = 100;
    const groupId = 10;
    const username = 'username';
    const groupName = 'group';

    const callbacks = {
        updateAclForUser: jest.fn(),
        updateAclForGroup: jest.fn(),
    };
    const props = {
        aclUsers: [{
            userId: userId,
            username: username,
            read: false,
            write: false,
            execute: false
        }],
        aclGroups: [{
            groupId: groupId,
            groupName: groupName,
            read: true,
            write: true,
            execute: true
        }],
        callbacks: callbacks
    };

    let component;

    beforeEach(() => {
        Object.entries(callbacks).forEach(([name, fn]) => {
            fn.mockClear();
        });
        component = shallow(<EditAcl {...props}/>);
    });

    it('match snapshot', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('updateAclForUser is called when clicking on user read permission checkbox', () => {
        component.find('[data-view="' + username + '-read"]')
            .simulate('change', {target: {name: 'read', checked: true}});

        expect(callbacks.updateAclForUser).toHaveBeenCalledWith(userId, 'read', true)
    });

    it('updateAclForGroup is called when clicking on group read permission checkbox', () => {
        component.find('[data-view="' + groupName + '-read"]')
            .simulate('change', {target: {name: 'read', checked: false}});

        expect(callbacks.updateAclForGroup).toHaveBeenCalledWith(groupId, 'read', false)
    });
});
