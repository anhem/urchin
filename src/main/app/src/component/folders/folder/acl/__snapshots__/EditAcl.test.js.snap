// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAcl match snapshot 1`] = `
<div
  className="edit-acl"
  data-view="editAcl"
>
  <h2>
    Group permissions
  </h2>
  <div
    className="edit-acl__groups"
  >
    <div
      className="edit-acl__groups__group"
      key="0"
    >
      <div
        data-view="group"
      >
        group
      </div>
      <div>
        <label>
          <input
            checked={true}
            data-view="group-read"
            name="read"
            onChange={[Function]}
            type="checkbox"
          />
          read
        </label>
      </div>
      <div>
        <label>
          <input
            checked={true}
            data-view="group-write"
            name="write"
            onChange={[Function]}
            type="checkbox"
          />
          write
        </label>
      </div>
      <div>
        <label>
          <input
            checked={true}
            data-view="group-execute"
            name="execute"
            onChange={[Function]}
            type="checkbox"
          />
          execute
        </label>
      </div>
    </div>
  </div>
  <h2>
    User permissions
  </h2>
  <div
    className="edit-acl__users"
  >
    <div
      className="edit-acl__users__user"
      key="0"
    >
      <div
        data-view="username"
      >
        username
      </div>
      <div>
        <label>
          <input
            checked={false}
            data-view="username-read"
            name="read"
            onChange={[Function]}
            type="checkbox"
          />
          read
        </label>
      </div>
      <div>
        <label>
          <input
            checked={false}
            data-view="username-write"
            name="write"
            onChange={[Function]}
            type="checkbox"
          />
          write
        </label>
      </div>
      <div>
        <label>
          <input
            checked={false}
            data-view="username-execute"
            name="execute"
            onChange={[Function]}
            type="checkbox"
          />
          execute
        </label>
      </div>
    </div>
  </div>
</div>
`;
